### PR TITLE
Sensitivity analysis

### PR DESCRIPTION
**Pull request by nataliecrowe060**
Sensitivity analysis finalized with additional text. Enjoyed trouble shooting the various issues and will try to continue to highlight any and all weak spots :)

### CODE REVIEW

File: Methane_MarkDown.Rmd [hunk 1]
```diff
@@ -111,3 +109,46 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+```
+
+Then we tell hector to run a range of values. In this case we ask for 20% above and bellow the set value of `NATURAL_CH4`(341Tg) at intervals of 5. 
+
+```{r sen-param}
+run_with_param_range <- function(core, parameter, values) {
+  mapped <- Map(function(x) param(core, parameter, x), values)
+  Reduce(rbind, mapped)
+}
+sensitivity_emission <- run_with_param_range(core, NATURAL_CH4(), seq(272.8, 358.05, 5))
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:140]: Clearer and better, I think, would be to

1. `fetchvars` the default value
2. Compute ±10% (or 20% or whatever)
3. Use those values in this `seq` call

File: Methane_MarkDown.Rmd [hunk 2]
```diff
@@ -111,3 +109,46 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+  setvar(core, NA, parameter, value, unit)
+  reset(core)
+  run(core)
+  result <- fetchvars(core, 2000:2100)
+  result[["parameter_value"]] <- value
+  result
+}
+```
+
+Then we tell hector to run a range of values. In this case we ask for 20% above and bellow the set value of `NATURAL_CH4`(341Tg) at intervals of 5.
```
Comment (bpbond) [inline Methane_MarkDown.Rmd:133]: Nice comment—clear and informative.

**Review by bpbond — APPROVED** [review]
Hi @nataliecrowe060 looks good! I have one suggestion, see below. Can you paste in the resulting graph from lines 145-151?

👏

File: Methane_MarkDown.Rmd (hunk 1 above)
Comment (nataliecrowe060) [reply Methane_MarkDown.Rmd:140]: What line would I add the change on? Also- it's Saturday thanks for the help I know it's not ideal! (Don't feel pressured to respond though if that was the extent on what you wanted to do today)

Comment (nataliecrowe060) [reply Methane_MarkDown.Rmd:140]: Also the graph you mention- it shows on my mark down, where you commenting to move it higher up?

**Comment by bpbond** [comment]
No I was asking you to paste the image into a comment here. Thanks!

________________________________

File: Methane_MarkDown.Rmd (hunk 1 above)
Comment (nataliecrowe060) [reply Methane_MarkDown.Rmd:140]: image

Comment (nataliecrowe060) [reply Methane_MarkDown.Rmd:140]: The graph ^

File: Methane_MarkDown.Rmd [hunk 3]
```diff
@@ -111,3 +109,46 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+  Reduce(rbind, mapped)
+}
+sensitivity_emission <- run_with_param_range(core, NATURAL_CH4(), seq(272.8, 358.05, 5))
+```
+
+Once we have our differing parameter values we can create a graph. Using the graph function `facet_wrap`to separate the results based on variable. 
+
+```{r sen-graph}
+ggplot(sensitivity_emission) +
+  aes(x = year, y = value, color = parameter_value, group = parameter_value) +
```
Comment (leeyap) [inline Methane_MarkDown.Rmd:147]: Nice graph! Maybe add a `linetype = units` or `shape = units` or something in your `aes()` call to have the units appear in the legend,

File: Methane_MarkDown.Rmd [hunk 4]
```diff
@@ -111,3 +109,46 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
...
+
+## Sensitivity Analysis
+
+To examine SSP245's Natural methane  sensitivity we first ask hector to fetch the data. We then create a function to run our parameter set to a particular value. In this case we set our value at `358.05`- twenty percent higher than the original value.
+
+```{r run-with-parameters}
+core <- newcore(ini_file)
+run(core)
+paramr<- fetchvars(core, NA, NATURAL_CH4())
+param<- function(core, parameter, value) {
```
Comment (leeyap) [inline Methane_MarkDown.Rmd:121]: Maybe add a comment here quickly explaining what this function does (same for line 136 below).

**Review by leeyap — APPROVED** [review]

**Pull request merged**